src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -653,6 +692,54 @@ public function resendEmailAction(int $orderId, int $orderStatusId, int $orderHi
         ]);
     }
 
+    public function deleteProductAction(int $orderId, int $orderDetailId): JsonResponse
+    {
+        $this->getCommandBus()->handle(
+            new DeleteProductFromOrderCommand($orderId, $orderDetailId)
+        );
+
+        return $this->json(null);
+    }
+
+    public function getPricesAction(int $orderId): JsonResponse
+    {
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+
+        return $this->json([
+            'orderTotalFormatted' => $orderForViewing->getPrices()->getTotalAmountFormatted(),
+            'productsTotalFormatted' => $orderForViewing->getPrices()->getProductsPriceFormatted(),
+            'shippingTotalFormatted' => $orderForViewing->getPrices()->getShippingPriceFormatted(),
+            'taxesTotalFormatted' => $orderForViewing->getPrices()->getTaxesAmountFormatted(),
+        ]);
+    }
+
+    /**
+     * Returns products for given order
+     *
+     * @param int $orderId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getProductsAction(int $orderId, Request $request): JsonResponse
+    {
+        $offset = $request->get('offset');
+        $limit = $request->get('limit');
+
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+
+        $products = $orderForViewing->getProducts()->getProducts();
+        if (null !== $limit && null !== $offset) {
+            $products = array_slice($products, (int) $offset, (int) $limit);
```

Comment:
So right now we load the full order in database, but only return some products, not all of them 🤔 .

I guess in the future we should optimize it, maybe by using a `GetPartialOrderForViewing` query which loads only the relevant products (offset and limits shall we put inside the query).
Can you please add a `@todo` statement here so we wont forget ?

Reply:
:ballot_box_with_check: Thanks